--- a/ext/gd/config.m4
+++ b/ext/gd/config.m4
@@ -135,7 +135,7 @@ AC_DEFUN([PHP_GD_XPM],[
   if test "$PHP_XPM_DIR" != "no"; then
 
     for i in $PHP_XPM_DIR /usr/local /usr/X11R6 /usr; do
-      test -f $i/$PHP_LIBDIR/libXpm.$SHLIB_SUFFIX_NAME || test -f $i/$PHP_LIBDIR/libXpm.a && GD_XPM_DIR=$i && break
+      test -f $i/$PHP_LIBDIR/libXpm.$SHLIB_SUFFIX_NAME || test -f $i/$PHP_LIBDIR/$DEB_HOST_MULTIARCH/libXpm.$SHLIB_SUFFIX_NAME || test -f $i/$PHP_LIBDIR/libXpm.a && GD_XPM_DIR=$i && break
     done
 
     if test -z "$GD_XPM_DIR"; then
@@ -208,6 +208,14 @@ AC_DEFUN([PHP_GD_FREETYPE2],[
         FREETYPE2_DIR=$i
         FREETYPE2_INC_DIR=$i/include/freetype2
         break
+      fi
+    done
+
+    for i in $PHP_FREETYPE_DIR /usr/local /usr; do
+      if test -f "$i/include/freetype2/freetype.h"; then
+        FREETYPE2_DIR=$i
+        FREETYPE2_INC_DIR=$i/include/freetype2
+        break
       fi
     done
 
