--- a/scripts/php-config.in
+++ b/scripts/php-config.in
@@ -2,6 +2,7 @@
 
 SED="@SED@"
 prefix="@prefix@"
+datarootdir="@datarootdir@"
 exec_prefix="@exec_prefix@"
 version="@PHP_VERSION@"
 vernum="@PHP_VERSION_ID@"
@@ -10,6 +11,7 @@ includes="-I$include_dir -I$include_dir/
 ldflags="@PHP_LDFLAGS@"
 libs="@EXTRA_LIBS@"
 extension_dir='@EXTENSION_DIR@'
+man_dir=`eval echo @mandir@`
 program_prefix="@program_prefix@"
 program_suffix="@program_suffix@"
 exe_extension="@EXEEXT@"
@@ -62,6 +64,8 @@ case "$1" in
   echo $php_sapis;;
 --configure-options)
   echo $configure_options;;
+--man-dir)
+  echo $man_dir;;
 --version)
   echo $version;;
 --vernum)
@@ -76,6 +80,7 @@ Options:
   --libs              [$libs]
   --extension-dir     [$extension_dir]
   --include-dir       [$include_dir]
+  --man-dir           [$man_dir]
   --php-binary        [$php_binary]
   --php-sapis         [$php_sapis]
   --phpapi            [$phpapi]
