--- a/ext/ming/config.m4
+++ b/ext/ming/config.m4
@@ -9,7 +9,7 @@ if test "$PHP_MING" != "no"; then
   AC_CHECK_LIB(m, sin)
 
   for i in $PHP_MING /usr/local /usr; do
-    if test -f $i/$PHP_LIBDIR/libming.$SHLIB_SUFFIX_NAME || test -f $i/$PHP_LIBDIR/libming.a; then
+    if test -f $i/$PHP_LIBDIR/libming.$SHLIB_SUFFIX_NAME || test -f $i/$PHP_LIBDIR/libming.a || test -f $i/lib/libming.$SHLIB_SUFFIX_NAME || test -f $i/lib/libming.$SHLIB_SUFFIX_NAME; then
       MING_DIR=$i
       break
     fi
